### PR TITLE
Cold staking

### PR DESCRIPTION
**Pull request by spazzymoto**
Draft pull request, i need particl-bitcore-lib to be merged so i can update the package-lock.json file

### CODE REVIEW

**Review by kewde — APPROVED** [review]
Looks alright to me!

File: lib/server.js [hunk 1]
```diff
@@ -3376,22 +3389,37 @@ WalletService.prototype.scan = function(opts, cb) {
...
+
+      var address = addresses.pop();
 
       checkActivity(wallet, address.address, function(err, activity) {
         if (err) return next(err);
 
         allAddresses.push(address);
-        inactiveCounter = activity ? 0 : inactiveCounter + 1;
+
+        if (activity)
```
Comment (rynomster) [inline lib/server.js]: Best to use our standards, always use `{` curly braces

File: lib/server.js [hunk 2]
```diff
@@ -3376,22 +3389,37 @@ WalletService.prototype.scan = function(opts, cb) {
...
+          activeDerivativeIndex = true;
+
+        if (_.isEmpty(addresses)) // Only increment when there is any activity on a address derived from this index
```
Comment (rynomster) [inline lib/server.js]: same here.. lets stick to the standards

File: lib/server.js [hunk 3]
```diff
@@ -2278,6 +2282,7 @@ WalletService.prototype.createTx = function(opts, cb) {
   opts = opts || {};
 
   function getChangeAddress(wallet, cb) {
+    // TODO: Wallets with single addresses not sha256 cannot cold stake
```
Comment (rynomster) [inline lib/server.js]: Is this just a reminder?

File: lib/model/wallet.js [hunk 4]
```diff
@@ -151,16 +151,32 @@ Wallet.prototype.isScanning = function() {
...
+
+  var self = this,
+      addresses = [];
+
+  var path = this.addressManager.getNewAddressPath(isChange);
+  log.verbose('Deriving addr:' + path);
+
+  addresses.push(Address.derive(self.id, this.addressType, this.publicKeyRing, path, this.m, this.coin, this.network, isChange));
+
+  if (self.coin === 'part')
```
Comment (rynomster) [inline lib/model/wallet.js]: Code standards `{`

**Review by rynomster — CHANGES_REQUESTED** [review]
Not sure if you ran the linter? or if there is one :P

**Comment by rynomster** [comment]
Doesn't look like there is a lint command

File: lib/server.js [hunk 5]
```diff
@@ -3376,33 +3388,53 @@ WalletService.prototype.scan = function(opts, cb) {
...
 
       checkActivity(wallet, address.address, function(err, activity) {
         if (err) return next(err);
 
         allAddresses.push(address);
-        inactiveCounter = activity ? 0 : inactiveCounter + 1;
 
-        if (!activity) 
+        if (activity) {
+          activeDerivativeIndex = true;
```
Comment (rynomster) [inline lib/server.js:3419]: Why not put an else return here? :)

**Pull request merged**